resources/views/dormitory/rooms/app.blade.php

```diff
@@ -124,6 +124,14 @@
                         array_push($colorArray, [ "id" => $room->name, "color" => $color]);
                     }
                     $roomColors=collect($colorArray);
+
+                    function getRoomColor(string|int $roomNumber, $roomColors) {
+                        return $roomColors->where('id',$roomNumber)->pluck('color')->join(',');
+                    }
+
+                    function getRoomResidents(string|int $roomNumber, \Illuminate\Database\Eloquent\Collection $rooms){
+                        return $rooms->firstWhere('name', $roomNumber)->users()->get()->pluck('name')->join(';');
```

Comment:
Do not add functions to blade files, please :D

Reply:
And where should they be then? Like this it does not look like shit :D

Reply:
I'd make an attribute getter for the room model that returns the colour accordingly and the names as a string. Then,  you can make one big array with different attributes in the controller and pass it to the view. I want to avoid using PHP in blades as much as possible